Makefile

```diff
@@ -11,17 +11,33 @@ cache_dir=$(deploy_dir)/cache
 
 config_file=_config.yml
 index_file=index.yml
+discover_config=_config/discover.yml
+update_config=_config/update.yml
+scrape_config=_config/scrape.yml
+search_config=_config/search_index.yml
+
+build: download-previous discover update scrape search-index
 
 # This target is invoked by a doc_independent job on the ROS buildfarm.
 html: build deploy
 
-# Clone a bunch of other repos part of the rosdistro and build the index.
-build:
+download-previous:
 	mkdir -p $(deploy_dir)
 	mkdir -p $(docs_dir)
 	mkdir -p $(remotes_dir)
 	vcs import --input $(remotes_file) --force $(remotes_dir)
-	bundle exec jekyll build --verbose --trace -d $(deploy_dir) --config=$(config_file),$(index_file)
+
+discover:
+	bundle exec jekyll build --verbose --trace -d $(deploy_dir) --config=$(config_file),$(index_file),$(discover_config)
+
+update:
+	bundle exec jekyll build --verbose --trace -d $(deploy_dir) --config=$(config_file),$(index_file),$(update_config)
+
+scrape:
+	bundle exec jekyll build --verbose --trace -d $(deploy_dir) --config=$(config_file),$(index_file),$(scrape_config)
+
+search-index:
+	bundle exec jekyll build --verbose --trace -d $(deploy_dir) --config=$(config_file),$(index_file),$(search_config)
```

Comment:
Just note that if any of these is run before `make download-previous`, `pulling_docs` fails because directories don't exist:

```
/mnt/TERA/code/rosindex/_plugins/pulling_docs.rb:16:in `block (2 levels) in <top (required)>': Unable to copy from /mnt/TERA/code/rosindex/_remotes/ros2_overview, the directory doesn't exist. (IOError)
```

It's not a big deal, I'm just pointing out in case we can avoid the error or improve the error message.

Reply:
Hm, good point. Ended up extending the error message so that it suggests to run the `download-previous` target.  thanks!

Reply:
Cool, but it looks like you haven't pushed the message change?

Reply:
Humm I see it now, sorry for the noise.

Reply:
Ah got it! Git(hub) always makes me feel like I'm going insane, if you go on the commit page https://github.com/ros2/rosindex/pull/67/commits/b4d7bfb541d8c4f701d8727620bd61e7c0d81b4a it says it was committed 2 hours ago :expressionless:

Reply:
Yeah, that happens because I "reused" the same commit (amended it) to include the missing content.
Given that the creation of the commit was indeed two hours ago, that timestamp didn't get updated after the amend. :)

Reply:
Thanks for the explanation. I'm still learning all the ways in which one can rewrite history with git, all of which feel criminal to me, since I'm used to mercurial :innocent: